src/main/java/com/pusher/client/PusherOptions.java

```diff
@@ -196,6 +198,26 @@ public String buildUrl(final String apiKey) {
                 : wsPort, apiKey, URI_SUFFIX);
     }
 
+    /**
+     *
+     * The default value is Proxy.NO_PROXY.
+     *
+     * @param proxy
+     *            Specify a proxy, e.g. <code>options.setProxy( new Proxy( Proxy.Type.HTTP, new InetSocketAddress( "proxyaddress", 80 ) ) )</code>;
+     * @return this, for chaining
+     */
+    public PusherOptions setProxy(Proxy proxy){
+        this.proxy = proxy;
```

Comment:
I'm pretty sure there will be an exception raised later if you set `null` here, so we should probably reject that early.